### PR TITLE
Initial Setup with config management.

### PR DESCRIPTION
**Pull request by RitikJaiswal75**
The intent of this pull request is to add support for running the backend in different environments, include Viper for multiple config management, add CORS support to the backend, and create a health endpoint.

- This pull request adds support for running the backend in different environments using command line arguments. This is useful because it allows you to easily switch between different environments (e.g. development, staging, production) without modifying your code.

- In addition to this, Viper has been included for multiple config management. This means that you can easily manage various configuration files for different environments.

- CORS support using gin/cors has also been added to the backend. This is important because it allows you to make cross-origin requests from your front end to your back end.

- Add support for integrating Firebase into our project. With this integration, we can now easily access our collections by simply calling their names. This makes it more convenient to work with data stored in Firebase collections.

- I have implemented and integrated a custom middleware into our system, which will effectively allow us to set up and configure authenticated routes for enhanced security and user authentication.

- Finally, a health endpoint has been created. This endpoint can be used to check the health of your backend and ensure that it is running correctly.

### CODE REVIEW

File: src/routes/routesHandler.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+import (
+	"github.com/Real-Dev-Squad/gopher-cloud-service/src/handlers"
+
+	"github.com/gin-gonic/gin"
+)
+
+func RoutesHandler(router *gin.Engine) {
+	router.GET("/health", handlers.HealthHandler)
+	router.GET("/health/dashboard", handlers.Dashboard)
```
Comment (prakashchoudhary07) [inline src/routes/routesHandler.go]: Why do you need a dashboard here?
What does it return?

File: src/main.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+func main() {
+	args := os.Args
+	var env string
+
+	if len(args) > 1 {
+		env = strings.Split(args[1], "=")[1]
+	} else {
+		env = "local"
+	}
+
```
Comment (prakashchoudhary07) [inline src/main.go:20]: Why are you reading it from args here?
Lets use ENV's here

File: src/handlers/dashboard.go [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+)
+
+func Dashboard(c *gin.Context) {
+	c.IndentedJSON(http.StatusOK, gin.H{"message": "you reached the dashboard"})
+}
```
Comment (prakashchoudhary07) [inline src/handlers/dashboard.go]: What is this doing? why do we need this?

**Review by prakashchoudhary07 — CHANGES_REQUESTED** [review]

File: src/main.go (hunk 2 above)
Comment (RitikJaiswal75) [reply src/main.go:20]: With reference to our communication over discord, I used this.
Link to the discord communication: https://discord.com/channels/673083527624916993/781622799868493864/1089968784904831040

File: src/routes/routesHandler.go (hunk 1 above)
Comment (RitikJaiswal75) [reply src/routes/routesHandler.go]: That is going to be an authenticated route. Also, I followed your PR reference where a monitor was set up
https://github.com/Real-Dev-Squad/gopher-cloud-service/pull/1/files#diff-20c10829adccb91d328e30581c58a0db895a36fa08653253978ea310755114deR20

File: src/handlers/dashboard.go (hunk 3 above)
Comment (RitikJaiswal75) [reply src/handlers/dashboard.go]: It has become an authenticated route

**Pull request closed without merging**